### PR TITLE
Run 'pub get' the first time the tests are run

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

File: bin/cache/README.md [hunk 1]
```diff
@@ -1,3 +0,0 @@
-# Binary cache
```
Comment (abarth) [inline bin/cache/README.md]: Did you delete this README file on purpose?

Comment (Hixie) [inline bin/cache/README.md]: fixed

File: packages/flutter_tools/lib/src/commands/test.dart [hunk 2]
```diff
@@ -47,9 +49,27 @@ class TestCommand extends FlutterCommand {
       testArgs.insert(0, '--no-color');
     List<BuildConfiguration> configs = buildConfigurations;
     bool foundOne = false;
+
+    if (!FileSystemEntity.isFileSync(path.join(flutterDir.path, 'pubspec.yaml'))) {
+      print('${flutterDir.path} has no pubspec.yaml');
+      return 1;
+    }
+
+    if (!FileSystemEntity.isFileSync(path.join(flutterDir.path, 'pubspec.lock'))) {
```
Comment (abarth) [inline packages/flutter_tools/lib/src/commands/test.dart]: You want to compare the mtime between these two files...

File: packages/flutter_tools/lib/src/commands/test.dart [hunk 3]
```diff
@@ -47,9 +49,27 @@ class TestCommand extends FlutterCommand {
...
+      print("Running pub get...");
+      Process process = await Process.start(
+        sdkBinaryName('pub'), ['get'],
+        workingDirectory: flutterDir.path
+      );
+      stdout.addStream(process.stdout);
+      stderr.addStream(process.stderr);
```
Comment (abarth) [inline packages/flutter_tools/lib/src/commands/test.dart]: I think process.dart has a wrapper that does this.  Something about "stream output"

**Comment by abarth** [comment]
LGTM

**Comment by Hixie** [comment]
@abarth ready for another look.

**Comment by abarth** [comment]
LGTM

**Pull request merged**